kubeadm: fix one missed pkg/errors location; drop pkg/errors from go.mod

neolit123 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?


/kind cleanup

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

- commit 1: PR #132080 removed the usage of pkg/errors in kubeadm, but missed one location.
- commit 2: drop pkg/errors as a direct dep in go.mod
- commit 3: forbids pkg/errors in k/k

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
xref #132080
xref #120276

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

neolit123 committed:
kubeadm: fix one missed pkg/errors location

Missed in PR 132080.

neolit123 commented:
/cc @SataQiu @HirazawaUi 

not sure how i missed this one. might have reverted it by mistake in between rebases.

HirazawaUi reviewed (commented):
/lgtm

Check again, only this one is missing
```
fgrep -r "github.com/pkg/errors" ./cmd/kubeadm
./cmd/kubeadm/app/cmd/phases/upgrade/apply/preflight.go:	"github.com/pkg/errors"
./cmd/kubeadm/app/util/errors/errors.go:// with stack traces similar to https://github.com/pkg/errors.
```

skitt committed:
Forbid github.com/pkg/errors in k/k

kubeadm has replaced pkg/errors with an internal implementation, which
means that pkg/errors can be marked as unwanted (once a final
straggler is replaced).

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented:
@neolit123 see https://github.com/kubernetes/kubernetes/pull/132126 for the missing pieces on the dependency side of things (that doesn’t have the preflight change just to check that the checks actually catch unwanted uses).

neolit123 committed:
go.mod: drop pkg/errors as a direct dep

neolit123 commented:
looks like we can drop pkg/errors as a direct dep from go.mod in this same PR (second commit).

/assign @dims
PTAL for approval

skitt commented:
@neolit123 there’s more to it than that, you need to update `hack/unwanted-dependencies.json` and you might as well then drop the pkg/errors checks in the linting config. Feel free to cherry-pick my commit :wink:

skitt commented:
And thanks for taking care of this @neolit123 :tada:!

dims commented:
/approve
/lgtm

thanks @skitt AND @neolit123
